### PR TITLE
CIS Authentication Origin Pull - Data Source files

### PR DESCRIPTION
**Pull request by arpit-srivastava-ibm**
Link to issue - https://github.ibm.com/NetworkTribe/nw-sdk-terraform-planner/issues/383
CIS Terraform Support for Authentication Origin Pull APIs
PR contains -

- Data Source file for Authentication Origin Pull APIs
- Test Cases for the data source file
- Documentation and Example for the Data source file

### CODE REVIEW

**Comment by arpit-srivastava-ibm** [comment]
Test Case Result - 

```
arpit-mac@Arpits-MacBook-Pro terraform-provider-ibm % make testacc TEST=./ibm/service/cis TESTARGS='-run=TestAccIBMCisOrigAuthDataSource_basic'
=== RUN   TestAccIBMCisOrigAuthDataSource_basic
--- PASS: TestAccIBMCisOrigAuthDataSource_basic (188.45s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cis	189.847s
```

**Comment by arpit-srivastava-ibm** [comment]
@kavya498 / @hkantare  We would need your review on this PR too. It is also a new feature - Authentication Origin Pull. 
We want to ship this in July's release, hence want you to review this by earliest.

**Comment by gahlaut-rahul** [comment]
@hkantare  Please review this and merge. It is also priority stuff.

File: ibm/service/cis/data_source_ibm_cis_origin_auth.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+				Description: "Associated CIS Request Type",
+				Optional:    true,
+				Default:     "zone_level",
+			},
+			"origin_pull_settings_enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "CIS origin auth settings enabled or disabled",
+			},
+			"origin_pull_certs_list": {
```
Comment (hkantare) [inline ibm/service/cis/data_source_ibm_cis_origin_auth.go:55]: No need to term the variable to include the type of variable by default indidcates its a list..Can you change this to "origin_pull_certs"

File: ibm/service/cis/data_source_ibm_cis_origin_auth.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+					Schema: map[string]*schema.Schema{
+						"cert_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certificate id",
+						},
+
+						"certificate": {
+							Type:        schema.TypeString,
+							Computed:    true,
```
Comment (hkantare) [inline ibm/service/cis/data_source_ibm_cis_origin_auth.go:69]: If this is sensitive value can you add Sensitive : true

File: ibm/service/cis/data_source_ibm_cis_origin_auth.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+						},
+						"cert_uploaded_on": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certifaate upldate time",
+						},
+						"cert_serial_number": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CIS origin auth certifaate upldate time",
```
Comment (hkantare) [inline ibm/service/cis/data_source_ibm_cis_origin_auth.go:100]: Description is wrong

File: ibm/service/cis/data_source_ibm_cis_origin_auth.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+	request_type := d.Get(CisOriginAuthRequestType).(string)
+
+	if request_type == "zone_level" {
+
+		// Get Zone Origin Pull Settings
+		zoneSettingsOpt := sess.NewGetZoneOriginPullSettingsOptions()
+		zoneSettingsResult, zoneSettingsResponse, zoneSettingsErr := sess.GetZoneOriginPullSettings(zoneSettingsOpt)
+
+		if zoneSettingsErr != nil || zoneSettingsResponse == nil {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error Getting Zone Level Origin Pull Settings: %s", zoneSettingsErr))
```
Comment (hkantare) [inline ibm/service/cis/data_source_ibm_cis_origin_auth.go:130]: Retrun the "zoneSettingsResponse" also along with the error in all return statements

File: website/docs/d/cis_auth_origin.html.markdown [hunk 5]
```diff
@@ -0,0 +1,49 @@
+---
+subcategory: "Internet services"
+layout: "ibm"
+page_title: "IBM: ibm_cis_origin_auths"
+description: |-
+  Get information on an IBM Cloud Internet Services Authentication Origin APIs.
+---
+
+# ibm_cis_alerts
```
Comment (hkantare) [inline website/docs/d/cis_auth_origin.html.markdown:9]: This is wrong

File: examples/ibm-cis/main.tf [hunk 6]
```diff
@@ -440,6 +440,19 @@ data "ibm_cis_alerts" "test1" {
...
+data "ibm_cis_origin_auths" "test" {
+	cis_id          = data.ibm_cis.cis.id
+	domain_id 		= data.ibm_cis_domain.cis_domain.domain_id
+}
+
+# CIS Authentication Origin Per Hostname Data source
+data "ibm_cis_origin_auths" "test" {
+	cis_id          = data.ibm_cis.cis.id
+	domain_id 		= data.ibm_cis_domain.cis_domain.domain_id
+  request_type    = "per_hostname"
```
Comment (hkantare) [inline examples/ibm-cis/main.tf:453]: Format the example

**Pull request merged**